src/main/java/jewellery/inventory/service/ProductInOrganizationService.java

```diff
@@ -298,6 +320,26 @@ private void moveQuantityFromResourcesInProductToResourcesInOrganization(Product
     product.setResourcesContent(null);
   }
 
+  private void updateProductOrganizationRecursively(Product product, Organization newOrganization) {
+    product.setOrganization(newOrganization);
+    logger.debug(
+        "Updated organization for product with ID: {}. New organization with ID: {}",
+        product.getId(),
+        newOrganization.getId());
+    if (product.getProductsContent() != null) {
+      List<Product> subProducts = product.getProductsContent();
+      for (Product subProduct : subProducts) {
+        updateProductOrganizationRecursively(subProduct, newOrganization);
+      }
+    }
+  }
+
+  private static void throwExceptionIfProductOrganizationEqualsRecipient(UUID recipientId, Product product) {
+    if (product.getOrganization().getId().equals(recipientId)) {
```

Comment:
This might throw an NullPointerException if the product is not owned by an organization.